### PR TITLE
FIX: catching exceptions does not work outside asyncio layer

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- Moved the generic block inside the asyncio layer of the wrapped call, and return the error message to the celery task to log it.
- catching `aiodocker.exceptions.DockerError` which was raised in called function
- moved `task_required_resources` error handling inside asyncio portion
- also fixes a test(`services/sidecar/tests/unit/test_rabbitmq.py`) which randomly fails, by giving the network more time to deliver a message

## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

**Comment by GitHK** [comment]
Adding more context.
Some pieces of code were not try catched for global failures while others where try catched outside of the async loop in the sync code. A try catch block which wraps around the `wrap_async_call` will not work.

File: services/sidecar/src/simcore_service_sidecar/core.py [hunk 1]
```diff
@@ -28,25 +30,34 @@
...
+                )
+                task = await result.fetchone()
+
+                if not task:
+                    log.warning("Task for node_id %s was not found", node_id)
+                    raise exceptions.TaskNotFound("Could not find a relative task")
+
+                # Image has to following format
+                # {"name": "simcore/services/comp/itis/sleeper", "tag": "1.0.0", "requires_gpu": false, "requires_mpi": false}
+                return {
```
Comment (pcrespov) [inline services/sidecar/src/simcore_service_sidecar/core.py:49]: This is a dict and does not correspond to the type ``bool``, annotated as the function returned type

File: services/sidecar/src/simcore_service_sidecar/celery_configurator.py [hunk 2]
```diff
@@ -94,12 +86,17 @@ def shared_task_dispatch(
         next_task_nodes = wrap_async_call(
             run_sidecar(celery_request.request.id, user_id, project_id, node_id)
         )
-        celery_request.update_state(state=states.SUCCESS)
 
+        if type(next_task_nodes) == str:
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/sidecar/src/simcore_service_sidecar/celery_configurator.py:90]: I think it migth be better that ``run_sidecar`` return a tuple such that
```python
new_task_nodes, error = run_sidecar(...)

if error:
    celery_request.update_state(state=states.FAILURE)
    return

celery_request.update_state(state=states.SUCCESS)
for _node_id in next_task_nodes:
   ...

```

it is similar to the *envelope* concept that we use in the rest API

File: services/sidecar/src/simcore_service_sidecar/celery_configurator.py [hunk 3]
```diff
@@ -40,15 +38,9 @@ def dispatch_comp_task(user_id: str, project_id: str, node_id: str) -> None:
...
-        required_resources = wrap_async_call(task_required_resources(node_id))
-    except Exception:  # pylint: disable=broad-except
-        log.error(
-            "%s\nThe above exception ocurred because it could not be "
-            "determined if task requires GPU or MPI for node_id %s",
-            traceback.format_exc(),
-            node_id,
-        )
+
+    required_resources = wrap_async_call(task_required_resources(node_id))
```
[the remark this answers is not in the document]
Comment (GitHK) [reply services/sidecar/src/simcore_service_sidecar/celery_configurator.py:42]: It's a bit more compact (not the obvious to me). It seems to work in the same manner.

File: services/sidecar/src/simcore_service_sidecar/celery_configurator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (GitHK) [reply services/sidecar/src/simcore_service_sidecar/celery_configurator.py:90]: > MINOR: ``isinstance(new_task_nodes, str)``` is the classic way to check types, right?

There is a slight difference between the two methods: `insistence` will be True also for subclasses, while `type` will only return the given type of the variable.
In this context it does not matter, both will work.

[the remark this answers is not in the document]
Comment (GitHK) [reply services/sidecar/src/simcore_service_sidecar/celery_configurator.py:90]: > I think it migth be better that `run_sidecar` return a tuple such that
> 
> ```python
> new_task_nodes, error = run_sidecar(...)
> 
> if error:
>     celery_request.update_state(state=states.FAILURE)
>     return
> 
> celery_request.update_state(state=states.SUCCESS)
> for _node_id in next_task_nodes:
>    ...
> ```
> 
> it is similar to the _envelope_ concept that we use in the rest API

Not bad, I like it more like this.

**Comment by GitHK** [comment]
fixes a few issues which could be liked to #1479

**Pull request merged**